### ISSUE TITLE
Details on latest commit/repo status when building doc

### ISSUE THREAD

**Issue by pomalley** [OddBloke/sphinx-git#20]
It's me again. :-)

I've added a `git_commit_detail` directive that can output the branch and link to GitHub commit, as well as warnings if there are uncommitted changes or untracked files at the time sphinx was run. I found this useful for my project, let me know what you think. (See my upcoming PR.)

**Comment by t2y** [comment]
I tried to use `git_commit_detail` with 6b11bd2b , but it causes an error in my environment.

* index.rst

```rest
.. git_commit_detail::                                                              
      :branch:                                                                        
      :commit:  
```

then, got an error.

```
    ...
    result = directive_instance.run()
  File "/home/tmorimoto/work/external/python/sphinx-git/sphinx_git/__init__.py", line 48, in run
    self.branch_name = self.repo.head.ref.name
  File "/home/tmorimoto/.virtualenvs/sphinx/lib/python2.7/site-packages/git/refs/symbolic.py", line 258, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to '7c9fd66fb28aaf7476cac3bf63907434dc4fc1c0'
```

My environment is here.

```shell
(sphinx)$ python --version
Python 2.7.11
(sphinx)$ pip freeze
GitPython==1.0.1
Sphinx==1.3.5
-e git+git@github.com:OddBloke/sphinx-git.git@6b11bd2b423588745203c08926bec408c1fada5a#egg=sphinx_git
```

**Comment by t2y** [comment]
hmm, I committed some changes into my repository. This error hasn't reproduced.

**Comment by pomalley** [comment]
OK, I'll admit I haven't tested it extensively; could it be that the repo was in detached head state? (That is, not on an actual branch?)

**Comment by OddBloke** [comment]
This was released in v0.10.0. Thanks @pomalley!
